packages/mongo/oplog_observe_driver.js

```diff
@@ -137,24 +137,37 @@ OplogObserveDriver = function (options) {
       var fence = DDPServer._CurrentWriteFence.get();
       if (!fence)
         return;
-      var write = fence.beginWrite();
-      // This write cannot complete until we've caught up to "this point" in the
-      // oplog, and then made it back to the steady state.
-      Meteor.defer(function () {
+
+      if (fence._oplogObserveDrivers) {
+        if (!_.contains(fence._oplogObserveDrivers, self))
+          fence._oplogObserveDrivers.push(self);
+
+        return;
+      }
+
+      fence._oplogObserveDrivers = [self];
```

Comment:
Not great to just put random fields on the fence.  But I can't immediately think of something better.